### ISSUE TITLE
Fail2ban: failed to ban IP

### ISSUE THREAD

**Issue by stephdl** [NethServer/dev#5503]
**Steps to reproduce**

* Fail2ban no warn if shorewall fails to ban/unban

our robb got some problem this weekend with fail2ban. https://community.nethserver.org/t/server-not-responsive-after-a-few-seconds/9890/

Robb played with openvpn with failed attempts, then fail2ban starts to ban his IP several time. I can see in his logs several ban for the same IP, each time we increased the counter of ban.
At the end the counter was 
```
213.49.181.131=ban
counter=17
date=Sat May 19 12:33:57 2018
```
This is impossible the counter cannot exceed `2` because if shorewall is workable, the IP cannot reach anymore the server

```
first jail ban -> counter 1 (+1) (IP banned, esmith key created)
recidive ban  -> counter  2 (+1) (after several ban, IP goes to recidive)
unban jail ban-> counter 1 (-1) (recidive still active)
unban recidive -> counter 0 (esmith key deleted and IP unbanned)
```

What I suspect is that shorewall stops and fail2ban spawn the command `shorewall drop <IP>` but we have no warn in logs if fail2ban bans with shorewall stopped **this my first bug**

* unban and delete the counter

When you do the unban, the counter is decreased but If the counter exceed `2` then the IP is not released and the key is not deleted because the counter is superior to zero.
We can admit if the counter is superior of `2` something was wrong and we can delete the key and unban the IP.

**Expected behavior**

I expect that 

- fail2ban die in log if it cannot ban or unban via shorewall 
- delete the e-smith key if the counter is superio to `2` when you do the unban

**Actual behavior**

- fail2ban doesn't warn in log when it cannot ban or unban
- the counter is increased and if the counter exceed `2` we cannot unban anymore an IP

**Components**

[root@prometheus ~]# rpm -qa |grep fail2ban
fail2ban-server-0.9.7-1.el7.noarch
fail2ban-firewalld-0.9.7-1.el7.noarch
fail2ban-shorewall-0.9.7-1.el7.noarch
nethserver-fail2ban-0.2.0-1.ns7.sdl.noarch
fail2ban-sendmail-0.9.7-1.el7.noarch
fail2ban-0.9.7-1.el7.noarch

----

thank robb

**Comment by stephdl** [comment]
testing case

- install the rpm
**- stop shorewall**
`systemctl stop shorewall`
- ban an IP manually
`/usr/libexec/nethserver/shorewall-nethserver  drop <ip> sshd`
- do it several 5 time or more
- check that the ip is banned 
`fail2ban-listban`
- You must find warns in /var/log/fail2ban/log
- the counter must be increased
`db fail2ban show <ip>`
- unban the IP manually 
`/usr/libexec/nethserver/shorewall-nethserver  allow <ip> sshd`
- check the counter, the key must have been removed and the e-smith key deleted
- start shorewall
`systemctl start shorewall`
- ban an ip
`/usr/libexec/nethserver/shorewall-nethserver  drop <ip> sshd`
- check there is no warns in log /var/log/fail2ban/log
- check that the ip is banned 
`fail2ban-listban`

**Comment by stephdl** [comment]
verified by mrmarcus, see  https://community.nethserver.org/t/server-not-responsive-after-a-few-seconds/9890/16

**Comment by stephdl** [comment]
closed, released as nethserver-fail2ban-1.0.2-1.ns7.noarch
